packages/react-router/lib/dom/ssr/browser.tsx

```diff
@@ -203,6 +262,17 @@ function createHydratedRouter(): RemixRouter {
   });
   ssrInfo.router = router;
 
+  // Do this after creating the router so ID's have been added to the routes
+  // that we can use as keys in the manifest. `router.routes` will contain the
+  // route IDs, `props.routes` may not.  If we encountered an error with the
+  // `propRoutes` above we'll have cleared them out so this is skipped
+  if (ssrInfo && propRoutes) {
+    let rootDataRoute = router.routes[0];
+    for (let route of rootDataRoute.children || []) {
+      addPropRoutesToRemix(ssrInfo, route as DataRouteObject, rootDataRoute.id);
+    }
+  }
```

Comment:
Add the new routes to the Remix `manifest`/`routeModules` structures as if they came from the server - they just won't have a `route.module`